ddtrace/tracer/span.go

```diff
@@ -205,7 +241,7 @@ func (s *span) Finish(opts ...ddtrace.FinishOption) {
 	}
 	if cfg.Error != nil {
 		s.Lock()
-		s.setTagError(cfg.Error, !cfg.NoDebugStack)
+		s.setTagError(cfg.Error, !cfg.NoDebugStack, cfg.StackFrames, cfg.SkipStackFrames)
```

Comment:
This is getting a bit ugly. We might want to think about maybe making this function take a unexported structure as a config:
```go
type errorConfig struct {
    debugStack  bool
    stackFrames uint
    skipFrames  uint
}
```
And then making it:
```go
func (s *span) setTagError(value interface{}, cfg *errorConfig) {
```

But I'm not convinced that this better. I'm just bringing it up for discussion.

Reply:
Please take a look how I solved it, not sure if it is much better.